lib/src/signed_video_h26x_sign.c

```diff
@@ -158,41 +174,35 @@ complete_sei_nalu_and_add_to_prepend(signed_video_t *self)
   SVI_DONE(status)
 
 done:
-  // Done with the SEI payload. Move |sei_data_buffer|. This should be done even if we caught a
-  // failure.
+
+  return status;
+}
+
+/* Removes the first element from the SEI buffer of a `signed_video_t` structure,
+ * shifting remaining elements left and clearing the last slot. */
+static void
+shift_sei_buffer_index(signed_video_t *self)
+{
+  const int sei_data_buffer_end = self->sei_data_buffer_idx;
   for (int j = 0; j < sei_data_buffer_end - 1; j++) {
     self->sei_data_buffer[j] = self->sei_data_buffer[j + 1];
   }
   self->sei_data_buffer[sei_data_buffer_end - 1].payload = NULL;
   self->sei_data_buffer[sei_data_buffer_end - 1].payload_signature_ptr = NULL;
   self->sei_data_buffer[sei_data_buffer_end - 1].last_two_bytes = LAST_TWO_BYTES_INIT_VALUE;
```

Comment:
What about the members `completed_sei` and `completed_sei_size`? You need to set/initialize them as well.